uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/contact-summary.js

```diff
@@ -1,62 +1,56 @@
-  return page.extend({
-
-    /**
-     * Opens the "contact access rights" modal
-     *
-     * @return {Promise} resolves with the modal page object
-     */
-    openManageRightsModal: function () {
-      var casper = this.casper;
-
-      return new Promise(function (resolve) {
-        casper.then(function () {
-          this.showActions();
-        }.bind(this));
-
-        casper.then(function () {
-          casper.click('[data-contact-access-rights]');
-          casper.waitWhileVisible('.spinner');
-
-          resolve(this.waitForModal('contact-access-rights'));
-        }.bind(this));
-      }.bind(this));
-    },
-
-    /**
-     * Opens one of the contact summary tabs
-     *
-     * @param  {string} tabId
-     * @return {object} resolves with the tab page object
-     */
-    openTab: function (tabId) {
-      var casper = this.casper;
-      var tab = require('./tabs/' + tabId).init(casper, false);
-
-      return new Promise(function (resolve) {
-        casper.then(function () {
-          casper.click('[title="' + tab.tabTitle + '"]');
-          casper.waitFor(tab.ready.bind(tab), function () {
-            casper.wait(500);
-            resolve(tab);
-          });
-        });
+module.exports = page.extend({
+  /**
+   * Opens the "contact access rights" modal
+   *
+   * @return {Promise} resolves with the modal page object
+   */
+  openManageRightsModal: function () {
+    var chromy = this.chromy;
+
+    return new Promise(function (resolve) {
+      this.showActions();
+
+      chromy.click('[data-contact-access-rights]');
+      chromy.wait(function () {
+        var dom = document.querySelector('.spinner');
+
+        return dom === null || (dom.offsetWidth <= 0 && dom.offsetHeight <= 0);
       });
-    },
 
-    /**
-     * Shows the dropdown of the "Actions" button in the contact summary page
-     */
-    showActions: function () {
-      var casper = this.casper;
-
-      casper.then(function () {
-        casper.click('#crm-contact-actions-link');
-        casper.waitUntilVisible('#crm-contact-actions-list');
-      });
-    }
-  });
-})();
+      resolve(this.waitForModal('contact-access-rights'));
+    }.bind(this));
+  },
+
+  /**
+   * Opens one of the contact summary tabs
+   *
+   * @param  {string} tabId
+   * @return {object} resolves with the tab page object
+   */
+  openTab: function (tabId) {
+    return new Promise(function (resolve) {
+      var tab = require('./tabs/' + tabId);
+      this.chromy.click('[title="' + tab.tabTitle + '"]');
+
+      resolve(tab.init(this.chromy, false));
+    }.bind(this));
+  },
+
+  /**
+   * Shows the dropdown of the "Actions" button in the contact summary page
+   */
+  showActions: function () {
+    this.chromy.click('#crm-contact-actions-link');
+    this.chromy.wait('#crm-contact-actions-list');
+  },
+
+  /**
+   * Wait an arbitrary amound of time for the data to load
+   */
+  waitForReady: function () {
+    this.chromy.wait(6000);
```

Comment:
Unfortunately we can't, the left hand side column of the contact summary has no spinner or anything that indicates that the data is loaded (except the data itself, but then we are relying on the data being there in the first place, which we should avoid if possible).

This really points more toward an improvement of the UX of the contact summary tab rather than an improvement of the backstopjs test (which would be improved automatically when the UX improves)